libnopegl/src/darray.c

```diff
@@ -128,13 +137,25 @@ void *ngli_darray_get(const struct darray *darray, size_t index)
     return darray->data + index * darray->element_size;
 }
 
+static void invalidate_range(struct darray *darray, size_t index, size_t count)
+{
+    if (!darray->user_free_func)
+        return;
+    const size_t end = index + count;
```

Comment:
You ll probably want the same check as it `remove_range`: `ngli_assert(end <= darray->count);`

Reply:
added